### DISCUSSION TITLE
Refining a specific region of an existing mesh

### DISCUSSION THREAD

**Discussion by iomags** [idaholab/moose#27314] [Q&A General]
I am trying to refine a specific region of an existing mesh (will eventually go for adaptive mesh refinement, so any tips on that appreciated, but for now attempting to do it manually) which has blocks and materials and everything already assigned. I have used SubdomainBoundingBoxGenerator to assign the regions in question to new block numbers and refine only those blocks, but they don't have any of the properties assigned to the original mesh. 

Essentially I have 2 touching blocks made from different materials, and I want to refine a volume that contains a small part of both blocks and the interface between them, so I want to define a new block containing only this region where the elements still have the same properties they'd have had before the new block existed.

Is there a way to transfer these properties automatically, or a better way to do this in general?

**Comment by GiudGiud** [comment]
Hello

Changing the subdomains will be problematic as material properties are assigned to a subdomain.

>  I want to refine a volume that contains a small part of both blocks and the interface between them, so I want to define a new block containing only this region where the elements still have the same properties they'd have had before the new block existed.

You could try:
- use two new subdomains for this purpose. One for each of the sub-parts of each block. And change the block-restriction of the original materials on the original blocks to include the new blocks. 
- use a different refinement technique. AMR is definitely a good way to do this. You could also try to refine the interface using a RefineSidesetGenerator?

**Reply by iomags** [reply]
Hi, 

Thank you for that, I have attempted to use AMR now and it seems to work initially but then the supercomputer kills it with signal 9 due to lack of memory. My first thought was to split the mesh to reduce the load, but does this refinement work with a pre-split mesh?

**Reply by GiudGiud** [reply]
hello

AMR works with pre-split meshes, no problem there
